app.js

```diff
@@ -76,6 +77,26 @@ app.param('build_id', function (req, res, next, buildId) {
   }
 })
 
+app.param('contentId', function (req, res, next, contentId) {
+  if (
+    contentId != null
+      ? contentId.match(OutputCacheManager.CONTENT_REGEX)
+      : undefined
+  ) {
+    return next()
+  } else {
+    return next(new Error(`invalid content id ${contentId}`))
```

Comment:
Would it be appropriate to throw a `Errors.NotFoundError` for these? It looks like the global error handler would then return a 404 instead of (AFAICT) a 5xx code for an invalid value.

Reply:
Side note: This code path is not used in production/SAAS.